aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1201,25 +1212,16 @@ impl AptosVM {
         G: AptosGasMeter,
         F: FnOnce(u64, VMGasParameters, StorageGasParameters, Gas) -> Result<G, VMStatus>,
     {
-        // TODO(Gas): revisit this.
-        let vm = AptosVM::new_from_state_view(state_view);
-
         // TODO(Gas): avoid creating txn metadata twice.
         let balance = TransactionMetadata::new(txn).max_gas_amount();
         let mut gas_meter = make_gas_meter(
-            vm.0.get_gas_feature_version(),
-            vm.0.get_gas_parameters(log_context)?.vm.clone(),
-            vm.0.get_storage_gas_parameters(log_context)?.clone(),
+            self.0.get_gas_feature_version(),
+            self.0.get_gas_parameters(log_context)?.vm.clone(),
+            self.0.get_storage_gas_parameters(log_context)?.clone(),
             balance,
         )?;
-
-        let resolver = StorageAdapter::new_with_cached_config(
-            state_view,
-            vm.0.get_gas_feature_version(),
-            vm.0.get_features(),
-        );
         let (status, output) =
-            vm.execute_user_transaction_impl(&resolver, txn, log_context, &mut gas_meter);
+            self.execute_user_transaction_impl(resolver, txn, log_context, &mut gas_meter);
```

Comment:
why does this need a resolver? (it has self)

Reply:
Not sure I follow how does self help here? The resolver is passed from outside?

Reply:
see my other comment, I think self may contain the resolver due to the caller.